retry_handler.go

```diff
@@ -144,6 +144,7 @@ func (middleware RetryHandler) retryRequest(ctx context.Context, pipeline Pipeli
 			ctx, span := otel.GetTracerProvider().Tracer(observabilityName).Start(ctx, "RetryHandler_Intercept - attempt "+fmt.Sprint(executionCount))
 			span.SetAttributes(attribute.Int("http.retry_count", executionCount),

```

Comment:
```suggestion
			span.SetAttributes(attribute.Int("http.request.resend_count", executionCount),

```

https://opentelemetry.io/docs/specs/semconv/http/http-spans/